packages/credential-provider-imds/src/fromContainerMetadata.ts

```diff
@@ -16,34 +16,33 @@ export const ENV_CMDS_AUTH_TOKEN = "AWS_CONTAINER_AUTHORIZATION_TOKEN";
  * Creates a credential provider that will source credentials from the ECS
  * Container Metadata Service
  */
-export function fromContainerMetadata(init: RemoteProviderInit = {}): CredentialProvider {
+export const fromContainerMetadata = (init: RemoteProviderInit = {}): CredentialProvider => {
   const { timeout, maxRetries } = providerConfigFromInit(init);
-  return () => {
-    return getCmdsUri().then((url) =>
-      retry(async () => {
-        const credsResponse = JSON.parse(await requestFromEcsImds(timeout, url));
-        if (!isImdsCredentials(credsResponse)) {
-          throw new ProviderError("Invalid response received from instance metadata service.");
-        }
-
-        return fromImdsCredentials(credsResponse);
-      }, maxRetries)
-    );
-  };
-}
+  return () =>
+    retry(async () => {
+      const requestOptions = await getCmdsUri();
```

Comment:
Moving `getCmdUri()` into function is fine as it only loads values from environmental variables.